torch/csrc/distributed/c10d/reducer.hpp

```diff
@@ -41,14 +57,45 @@ class TORCH_API Timer {
     kBackwardCommEnd,
   };
 
-  // Record the current event, i.e., mark it as having occurred now.
-  virtual void record(Event event) = 0;
+  // Record the current event, i.e., mark it as having occurred now. Default
+  // CPU implementation.
+  virtual void record(Event event) {
+    getTime(event) = current_time_in_nanos();
```

Comment:
Nit: Better to rename it as `getTimeRef`, since this method is not a common read-only getter, but returns a mutable reference.